oauth2_http/javatests/com/google/auth/oauth2/ITWorkloadIdentityFederationTest.java

```diff
@@ -0,0 +1,261 @@
+
+    JsonObjectParser parser = new JsonObjectParser(GsonFactory.getDefaultInstance());
+    request.setParser(parser);
+
+    HttpResponse response = request.execute();
+    String rawXml = response.parseAsString();
+
+    String awsAccessKeyId = getXmlValueByTagName(rawXml, "AccessKeyId");
+    String awsSecretAccessKey = getXmlValueByTagName(rawXml, "SecretAccessKey");
+    String awsSessionToken = getXmlValueByTagName(rawXml, "SessionToken");
+
+    AwsCredentials awsCredentials =
+        (AwsCredentials)
+            AwsCredentials.fromJson(buildAwsCredentialConfig(), OAuth2Utils.HTTP_TRANSPORT_FACTORY);
+
+    environmentVariables.set("AWS_ACCESS_KEY_ID", awsAccessKeyId);
+    environmentVariables.set("AWS_SECRET_ACCESS_KEY", awsSecretAccessKey);
+    environmentVariables.set("Token", awsSessionToken);
+    environmentVariables.set("AWS_REGION", "us-east-2");
+
+    callGcs(awsCredentials);
+  }
+
+  private GenericJson buildIdentityPoolCredentialConfig() throws IOException {
+    String idToken = generateGoogleIdToken(OIDC_AUDIENCE);
+
+    File file =
+        File.createTempFile(
+            "ITWorkloadIdentityFederation", /* suffix= */ null, /* directory= */ null);
+    file.deleteOnExit();
+    OAuth2Utils.writeInputStreamToFile(
+        new ByteArrayInputStream(idToken.getBytes(StandardCharsets.UTF_8)), file.getAbsolutePath());
+
+    GenericJson config = new GenericJson();
+    config.put("type", "external_account");
+    config.put("audience", OIDC_AUDIENCE);
+    config.put("subject_token_type", "urn:ietf:params:oauth:token-type:jwt");
+    config.put("token_url", "https://sts.googleapis.com/v1beta/token");
+    config.put(
+        "service_account_impersonation_url",
+        String.format(
+            "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/%s:generateAccessToken",
+            clientEmail));
+
+    GenericJson credentialSource = new GenericJson();
+    credentialSource.put("file", file.getAbsolutePath());
+    config.put("credential_source", credentialSource);
+
+    return config;
+  }
+
+  private GenericJson buildAwsCredentialConfig() {
+    GenericJson config = new GenericJson();
+    config.put("type", "external_account");
+    config.put("audience", AWS_AUDIENCE);
+    config.put("subject_token_type", "urn:ietf:params:aws:token-type:aws4_request");
+    config.put("token_url", "https://sts.googleapis.com/v1beta/token");
+    config.put(
+        "service_account_impersonation_url",
+        String.format(
+            "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/%s:generateAccessToken",
+            clientEmail));
+
+    GenericJson credentialSource = new GenericJson();
+    credentialSource.put("environment_id", "aws1");
+    credentialSource.put(
+        "regional_cred_verification_url",
+        "https://sts.{region}.amazonaws.com?Action=GetCallerIdentity&Version=2011-06-15");
+    config.put("credential_source", credentialSource);
+
+    return config;
+  }
```

Comment:
If this is static content, can we put a json file in test/resources and load the file?

Reply:
Yes, but I did it this way because if we were to run the tests on different inputs (pool, provider, service account), we'd have to make changes in 2 places vs just 1 now.